src/test/text_streamer_test.cpp

```diff
@@ -0,0 +1,132 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <string>
+
+#include "../llm/llmnoderesources.hpp"
+#include "gtest/gtest.h"
+#include "test_utils.hpp"
+
+class TextStreamerTest : public ::testing::Test {
+public:
+    ::mediapipe::CalculatorGraphConfig config;
+    std::shared_ptr<ovms::LLMNodeResources> nodeResources = nullptr;
+    std::shared_ptr<Tokenizer> tokenizer;
+    std::shared_ptr<ovms::TextStreamer> streamer;
+    std::string testPbtxt = R"(
+    node: {
+    name: "llmNode"
+    calculator: "HttpLLMCalculator"
+    node_options: {
+        [type.googleapis.com / mediapipe.LLMCalculatorOptions]: {
+            models_path: "/ovms/llm_testing/facebook/opt-125m"
+        }
+    }
+    }
+)";
+
+    void SetUp() override {
+        py::initialize_interpreter();
+        ASSERT_TRUE(::google::protobuf::TextFormat::ParseFromString(testPbtxt, &config));
+        ASSERT_EQ(ovms::LLMNodeResources::createLLMNodeResources(nodeResources, config.node(0), ""), ovms::StatusCode::OK);
+        tokenizer = std::make_shared<Tokenizer>("/ovms/llm_testing/facebook/opt-125m");
+        streamer = std::make_shared<ovms::TextStreamer>(tokenizer);
+    }
+    void TearDown() {
+        nodeResources.reset();
+        py::finalize_interpreter();
+    }
+};
+
+TEST_F(TextStreamerTest, noValueReturned) {
+    auto tokens = tokenizer->encode("TEST");
+    for (size_t i = 0; i < tokens.get_size(); i++) {
```

Comment:
We have no control over the tokenizer, if at some point in the future the tokenizer will change its behavior to return empty vector, this test will **pass**, but in fact it will not test your intended use case.

To protect against that I would suggest to provide tokens yourself instead of tokenizing the text. (You can use the tokenizer to get the tokens, but assert that those tokens are exactly the same as you expect before running the test)

The same comment can be applied for all the unit tests in this PR
Please let me know your opinion about this

Reply:
Once we have that in test it will be easier to read it, otherwise the reader never knows how many times the for loop is expected to run